api/medical/views.py

```diff
@@ -344,17 +344,47 @@ def get(self, request):
 
     def post(self, request):
         patient = request.POST.get("patient", None)
-        patientinstance = User.objects.filter(id=patient).first()
+        patientinstance = User.objects.filter(email=patient).first()
         doctor = request.POST.get("doctor", None)
         doctorinstance = User.objects.filter(id=doctor).first()
         details = request.POST.get("details", None)
         date = request.POST.get("date", None)
+        inhouse = request.POST.get("inhouse", None)
+        diagnosis = request.POST.get("diagnosis", None)
+        treatment = request.POST.get("treatment", None)
+        time = request.POST.get("time", None)
+
         instance = MedicalHistory.objects.create(
             patient=patientinstance,
             date=date,
             doctor=doctorinstance,
-            details=details
+            details=details,
+            inhouse=inhouse,
+            diagnosis=diagnosis,
+            treatment=treatment,
+            time=time
         )
         instance.save()
         serinstance = MedicalHistorySerializer(instance)
         return Response(serinstance.data)
+
+class SearchPatient(APIView):
+    permission_classes = (IsAuthenticated,)
```

Comment:
This API is for medical manager, hence use `IsMedicalManager` instead of `IsAuthenticated`.

Reply:
done